### PR TITLE
1 a 2

### PR DESCRIPTION
**Pull request by alicampbell2**

### CODE REVIEW

**Comment by alicampbell2** [comment]
Reminder to check all libraries are called correctly for functions used (e.g. DT for datatable)

**Review by abhsheksingh — CHANGES_REQUESTED** [review]
[comment]: <> (https://code.visualstudio.com/docs/languages/markdown#_markdown-preview)

[comment]: <> (This is a comment, it will not be included)

##  Code review by Abhishek Singh

#### Documentation


  
- [x] ReadMe for the indicator

    1.	Instructions numbering is not in correct order.
    2.	Very clear instructions. 


- [ ] example_config file for the indicator

    1.	Edit line 1 to read “# configurations for indicator 1-a-2”
    2.	Remove the generic hashed out lines (4-8):
        - ##If you are using csv data make sure numbers are formatted as numbers in Excel
        -	##and DO NOT include the thousands separator
        - ##tabname will not be needed if you are using csv data
        - #filename <- "type_1_metadata_included.csv"
        - #filename <- "type_1_no_metadata.csv"


- [x] compile_tables file for the indicator 

- [x] Example input and output that run successfully locally. 


  

#### Functionality

- [x] **Installation**: Installation or build of the code succeeds. 
- [x] **Guidelines**: New code conforms to the project contribution guidelines. 
- [x] **Example data**: Example output generated using the example input matches the given example output. 
- [x] **csv formatting**: The csv output is correct, given general guidance for csvs (e.g. columns are in 
a logical order, values within columns are in the correct order, sentence case is used, correct terms are 
used (i.e. ar in the data dictionary), observation status column is correctly filled in).  

     

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

Estimated time spent reviewing: 2 hrs

---

### Review comments




1. ReadMe for the indicator

    - 	Instructions numbering is not in correct order ***(The numbering will help in clearly following the instructions ).***
    - 	Although very clear instructions written. 


2. example_config file for the indicator (Minor changes)

    -	Edit line 1 to read “# configurations for indicator 1-a-2” ***(for clarity)***
    -	Remove the generic hashed out lines (4-8) ***(Not importnat! but if you wish to remove )*** :
        - ##If you are using csv data make sure numbers are formatted as numbers in   Excel
        -	##and DO NOT include the thousands separator
        - ##tabname will not be needed if you are using csv data
        - #filename <- "type_1_metadata_included.csv"
        - #filename <- "type_1_no_metadata.csv"

3. I have got this warning message in console, while running the code for the first time, after that it was fine ***(Putting in here for documentation).***

    Warning message:
      - Using an external vector in selections was deprecated in tidyselect 1.1.0.
      - 1 Please use `all_of()` or `any_of()` instead.
      - #Was: data %>% select(most_recent_year)

      - #Now: data %>% select(all_of(most_recent_year))

**Comment by alicampbell2** [comment]
Addressed the comments on ReadMe and config, not entirely sure what the warning refers to as all_of is used in the code

**Comment by abhsheksingh** [comment]
> Addressed the comments on ReadMe and config, not entirely sure what the warning refers to as all_of is used in the code

Its fine @alicampbell2  as it might be warning from my system as I started new, there might be some package or something.

File: 1-a-2/1-a-2_README.md [hunk 1]
```diff
@@ -14,9 +14,9 @@ This is a very simple automation: we read in one data table, calculate a percent
   - You WILL need to change the most_recent_year to be accurate
```
Comment (abhsheksingh) [inline 1-a-2/1-a-2_README.md:14]: Copy the text below for the Readme file if you like the indentation, or you can leave it as it is (its fine).
### 1-a-2 automation
  
This is a very simple automation: we read in one data table, calculate a percentage from two of the rows and output one value per year.
  
### Instructions to run update ###
1. *UK SDG data team:* (once this code has been reviewed and is on the main branch) Go to Jemalex > sdg_data_updates.    
   *Others:* Checkout sdg_data_updates main branch to your local repository.
2. Download the data from https://www.gov.uk/government/statistics/public-expenditure-statistical-analyses-2022 (PESA [year] Chapter 4 Tables > Table 4_2) and place it in the Input folder in sdg_data_updates > 1-a-2. Make sure the filename is in the format "PESA_2022_CP_Chapter_4_tables.xlsx". There should only be one file in the input folder - the most recent tables.
3. Open sdg_data_updates.Rproj  
4. Change indicator folder name (`indicator <- "1-a-2"`)  
5. If config.R does not exist in the 1-a-2 folder, create it from the example_config.R file  
6. Check the configs are correct:
    - You may need to change tabname and header_row if these have changed between years
    - You WILL need to change the most_recent_year to be accurate

      To run the example data and get example output, it should be example_config.R.   
7. Open update_indicator_main.R 
    - Make sure test_run is TRUE or FALSE: changes which Input folder is used
8. Click Source (by default this is in top right of the script window)  
9. Check for messages in the console. When the script is run a file titled '1-a-2.csv' will be saved in 1-a-2 > Output 
   Use this file for the Indicator csv.  
10. A file called <date>_1-a-2_checks.html will also be in the outputs folder. Read through this as a QA of the csv.

**Review by abhsheksingh — APPROVED** [review]
All looks good!

**Pull request merged**